src/store/reducers/user.ts

```diff
@@ -1,14 +1,22 @@
 import { Reducer } from "redux";
 import { User } from "../../models/user";
-import { UserAction, UserActionType } from "../actions/types";
+import { UserAction, UserActionType, WebSocketAction, WebSocketActionType } from "../actions/types";
 
-export const userReducer: Reducer<User | null, UserAction> = (user = null, action) => {
+export const userReducer: Reducer<User | null, UserAction | WebSocketAction> = (user = null, action) => {
     switch(action.type) {
         case UserActionType.SET:
             return {...action.user};
     }
     if(user === null) {
         return null;
     }
+
+    switch(action.type) {
```

Comment:
Uhhh, a switch statement with a single case... Isn't that just like an if-statement lol. I would replace it with an if statement, then you could also merge it with the if statement that is already in the case.